### PR TITLE
Not relying on GetAvailableTypes for message endpoint mapping

### PR DESCRIPTION
**Pull request by danielmarbach**
Relates to https://github.com/Particular/NServiceBus/pull/4084

Addresses https://github.com/Particular/NServiceBus/pull/4084#discussion_r76755596

@SzymonPobiega @Particular/nservicebus-maintainers thoughts?

### CODE REVIEW

**Comment by andreasohlund** [comment]
LGTM

File: src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs [hunk 1]
```diff
@@ -209,5 +207,33 @@ static void RegisterMessageRoute(Type mappedType, string address, Dictionary<Typ
             }
             routeTableEntries[mappedType] = new RouteTableEntry(mappedType, route);
         }
+
+        static List<Type> GetBaseTypes(Type messageType, Conventions conventions)
+        {
+            var result = new List<Type>();
+            var baseType = messageType.BaseType;
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs]: Is it enough to walk the base types of do we need to look at implemented interfaces as well?

Comment (danielmarbach) [inline src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs]: @andreasohlund see https://github.com/Particular/NServiceBus/pull/4097/files/82142c44bd97b6a6516123c17619ed8746194552#diff-d9f6e42c7a4ae0af9eb4c474ce658281R228 I get the implemented interfaces. Isn't that enough? At least serializer does it that way as well

Comment (andreasohlund) [inline src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs]: yup, I see that now :+1:

**Comment by andreasohlund** [comment]
All tests added by @SzymonPobiega are still green so I'm going to merge this

**Comment by SzymonPobiega** [comment]
Good stuff @danielmarbach . Thanks!

**Pull request merged**